tools/mwa/mwa_astro_tool.xml

```diff
@@ -0,0 +1,117 @@
+<tool id="mwa_astro_tool" name="MWA" version="0.0.4+galaxy0" profile="23.0">
+  <requirements>
+    <requirement type="package" version="8.20.0">ipython</requirement>
+    <requirement type="package" version="5.3.4">astropy</requirement>
+    <requirement type="package" version="1.1">gammapy</requirement>
+    <requirement type="package" version="3.8.2">matplotlib</requirement>
+    <requirement type="package" version="0.4.6">astroquery</requirement>
+    <requirement type="package" version="1.12.0">scipy</requirement>
+    <requirement type="package" version="1.2.7">oda-api</requirement>
+    <requirement type="package" version="7.14.2">nbconvert</requirement>
+    <requirement type="package" version="1.20.3">wget</requirement>
+    <!--Requirements string 'nb2workflow[cwl,service,rdf,mmoda]>=1.3.30 
+' can't be converted automatically. Please add the galaxy/conda requirement manually or modify the requirements file!-->
+  </requirements>
+  <command detect_errors="exit_code">ipython '$__tool_directory__/${_data_product._selector}.py'</command>
+  <configfiles>
+    <inputs name="inputs" filename="inputs.json" />
+  </configfiles>
+  <inputs>
+    <conditional name="_data_product">
+      <param name="_selector" type="select" label="Data Product">
+        <option value="Image" selected="true">Image</option>
+        <option value="Spectrum" selected="false">Spectrum</option>
+      </param>
+      <when value="Image">
+        <param name="src_name" type="text" value="1ES 0229+200" label="src_name" />
+        <param name="RA" type="float" value="38.202562" label="RA Units: deg" />
+        <param name="DEC" type="float" value="20.288191" label="DEC Units: deg" />
+        <param name="T1" type="text" value="2000-10-09T13:16:00.0" label="T1" />
+        <param name="T2" type="text" value="2022-10-10T13:16:00.0" label="T2" />
+        <param name="Radius" type="float" value="1.0" label="Radius Units: deg" />
+        <param name="pixsize" type="float" value="0.01" label="pixsize Units: deg" />
+        <param name="Frequency" type="select" label="Frequency">
+          <option value="GLEAM 103-134 MHz">GLEAM 103-134 MHz</option>
+          <option value="GLEAM 139-170 MHz">GLEAM 139-170 MHz</option>
+          <option value="GLEAM 170-231 MHz" selected="true">GLEAM 170-231 MHz</option>
+          <option value="GLEAM 72-103 MHz">GLEAM 72-103 MHz</option>
+        </param>
+      </when>
+      <when value="Spectrum">
+        <param name="src_name" type="text" value="1ES 0229+200" label="src_name" />
+        <param name="RA" type="float" value="38.202562" label="RA Units: deg" />
+        <param name="DEC" type="float" value="20.288191" label="DEC Units: deg" />
```

Comment:
just a very silly question ... is `deg` degree? And do you think it would make sense to have a type="deg"? What would such a type have as properties?

You could also add a min="" and max="" here I think.

Reply:
What we are currently dealing with, out of galaxy context, is the structured input type, e.g. a table that gets encoded as a dict of lists. At some point, we will need to find a way to have such a form of input in galaxy. So we will most likely try to extend this part